Packs/FeedUnit42v2/Integrations/FeedUnit42v2/FeedUnit42v2.py

```diff
@@ -0,0 +1,540 @@
+            "type": 'Attack Pattern',
+            "score": 0,
+            "fields": {
+                'stixid': coa_indicator.get('id'),
+                "firstseenbysource": handle_multiple_dates_in_one_field('created', coa_indicator.get('created')),
+                "modified": handle_multiple_dates_in_one_field('modified', coa_indicator.get('modified')),
+                'description': coa_indicator.get('description', ''),
+                "publications": publications,
+                "reportedby": 'Unit42',
+                "tags": feed_tags,
+            }
+        }
+        if tlp_color:
+            indicator['fields']['trafficlightprotocol'] = tlp_color
+
+        course_of_action_indicators.append(indicator)
+    return course_of_action_indicators
+
+
+def get_ioc_type(indicator, id_to_object):
+    """
+    Get IOC type by extracting it from the pattern field.
+
+    Args:
+        indicator: the indicator to get information on.
+        id_to_object: a dict in the form of - id: stix_object.
+
+    Returns:
+        str. the IOC type.
+    """
+    ioc_type = ''
+    indicator_obj = id_to_object.get(indicator, {})
+    pattern = indicator_obj.get('pattern', '')
+    for unit42_type in UNIT42_TYPES_TO_DEMISTO_TYPES:
+        if unit42_type in pattern:
+            ioc_type = UNIT42_TYPES_TO_DEMISTO_TYPES.get(unit42_type)  # type: ignore
+            break
+    return ioc_type
+
+
+def get_ioc_value(ioc, id_to_obj):
+    """
+    Get IOC value from the indicator name field.
+
+    Args:
+        ioc: the indicator to get information on.
+        id_to_obj: a dict in the form of - id: stix_object.
+
+    Returns:
+        str. the IOC value. if its reports we add to it [Unit42 ATOM] prefix.
+    """
+    ioc_obj = id_to_obj.get(ioc)
+    if ioc_obj:
+        return f"[Unit42 ATOM] {ioc_obj.get('name')}" if ioc_obj.get('type') == 'report' else ioc_obj.get('name')
+
+
+def create_list_relationships(relationships_objects, id_to_object):
+    """Parse the Relationships objects retrieved from the feed.
+
+    Args:
+      relationships_objects: a list of relationships objects containing the relationships.
+      id_to_object: a dict in the form of - id: stix_object.
+
+    Returns:
+        A list of processed relationships.
+    """
+    relationships_list = []
+
+    for relationships_object in relationships_objects:
+
+        relationship_type = relationships_object.get('relationship_type')
+        if relationship_type not in RELATIONSHIP_TYPES:
+            if relationship_type == 'indicates':
+                relationship_type = 'indicated-by'
+            else:
+                demisto.debug(f"Invalid relation type: {relationship_type}")
+                continue
+
+        a_type = relationships_object.get('source_ref').split('--')[0]
+        a_type = RELATIONS_TYPE_TO_DEMISTO_TYPES.get(a_type)
+        if a_type == 'Indicator':
+            a_type = get_ioc_type(relationships_object.get('source_ref'), id_to_object)
```

Comment:
think it is better to call it as source or what ever it is, and assign it to a_type only in the end when it is actually the type at that point.
this is relevant also to the B type code after this comment